src/TaskRunner/Commands/BuildCommands.php

```diff
@@ -46,6 +46,10 @@ public function npmRun(string $npmCommand, array $options = [
       ->dir($npmDir)
       ->stopOnFail();
 
+    if (!file_exists($npmDir . '/package-lock.json')) {
```

Comment:
What is the scenario on this? The file gets removed if a new version of the theme is downloaded? Cause dependencies might be updated, so the lock is outdated at that moment.

Reply:
I think we should probably commit the `package-lock.json` file and update dependencies periodically. This makes it possible for the theme developers to install a working version of the development environment, also for historic versions.

I am struggling right now with a website that doesn't commit the lock file. The theme is now 2 years old and it is practically impossible to figure out which versions I need to install to make the gulp commands work. I get recent versions of the dependencies and now the gulp commands are throwing errors, and now I need to do detective work in changelogs to find out which exact versions I need for these commands to work. If the lock file was committed, then I would get the exact versions and I would be able to compile the SASS and minify assets.

I think the main idea is:
* Production code of libraries / modules / plugins: do not commit lock file, the composer / package files should declare _the range of versions the library is compatible with_.
* Development code of libraries (and production code of actual projects): commit lock files.

So in the case of the OE Bootstrap theme: we should not commit `composer.lock` (production code) but we should commit `package-lock.json` (dev environment).